pkg/rpcclient/nns/contract_test.go

```diff
@@ -195,3 +224,180 @@ func TestGetAllRecordsExpanded(t *testing.T) {
 		Data: "cool",
 	}, vals[0])
 }
+
+func TestRoots(t *testing.T) {
+	ta := &testAct{}
+	nns := NewReader(ta, util.Uint160{1, 2, 3})
+
+	iid := uuid.New()
+
+	// Session-based iterator.
+	sid := uuid.New()
+	ta.res = &result.Invoke{
+		Session: sid,
+		State:   "HALT",
+		Stack: []stackitem.Item{
+			stackitem.NewInterop(result.Iterator{
+				ID: &iid,
+			}),
+		},
+	}
+	iter, err := nns.Roots()
+	require.NoError(t, err)
+
+	ta.res = &result.Invoke{
+		Stack: []stackitem.Item{
+			stackitem.Make([]stackitem.Item{
+				stackitem.Make("n3"),
+				stackitem.Make("aaaaaa"),
+				stackitem.Make("cool"),
+			}),
+		},
+	}
+	vals, err := iter.Next(10)
+	require.NoError(t, err)
+	require.Equal(t, 1, len(vals))
+	require.Equal(t, "n3", vals[0])
+
+	ta.err = errors.New("")
+	_, err = iter.Next(1)
+	require.Error(t, err)
+
+	err = iter.Terminate()
+	require.Error(t, err)
+
+	// Value-based iterator.
+	ta.err = nil
+	ta.res = &result.Invoke{
+		State: "HALT",
+		Stack: []stackitem.Item{
+			stackitem.NewInterop(result.Iterator{
+				Values: []stackitem.Item{
+					stackitem.Make("n3"),
+					stackitem.Make("aaaaaa"),
+					stackitem.Make("cool"),
+				},
+			}),
+		},
+	}
+	iter, err = nns.Roots()
+	require.NoError(t, err)
+
+	ta.err = errors.New("")
+	err = iter.Terminate()
+	require.NoError(t, err)
+}
+
+func TestUpdate(t *testing.T) {
+	ta := &testAct{}
+	nns := New(ta, util.Uint160{1, 2, 3})
+
+	nef := []byte{0x01, 0x02, 0x03}
+	manifest := "manifest data"
+
+	ta.err = errors.New("test error")
+	_, _, err := nns.Update(nef, manifest)
+	require.Error(t, err)
+
+	// Test successful update
+	ta.err = nil
+	ta.txh = util.Uint256{0x04, 0x05, 0x06}
+	txh, vub, err := nns.Update(nef, manifest)
+	require.NoError(t, err)
+	require.Equal(t, ta.txh, txh)
+	require.Equal(t, ta.vub, vub)
+
+	for _, fun := range []func(nef []byte, manifest string) (*transaction.Transaction, error){
+		nns.UpdateTransaction,
+		nns.UpdateUnsigned,
+	} {
+		ta.err = errors.New("")
+		_, err := fun(nil, "")
+		require.Error(t, err)
+
+		ta.err = nil
+		ta.tx = &transaction.Transaction{Nonce: 100500, ValidUntilBlock: 42}
+		tx, err := fun(nil, "")
+		require.NoError(t, err)
+		require.Equal(t, ta.tx, tx)
+	}
+}
+
+func TestAddRoot(t *testing.T) {
+	ta := &testAct{}
+	nns := New(ta, util.Uint160{1, 2, 3})
+
+	root := "example.root"
+	params, err := smartcontract.NewParameterFromValue(root)
+	require.NoError(t, err)
+	ta.err = errors.New("test error")
+	_, _, err = nns.AddRoot(params.Value.(string))
+	require.Error(t, err)
+
+	// Test success case
+	ta.err = nil
+	ta.txh = util.Uint256{0x07, 0x08, 0x09}
+	txh, vub, err := nns.AddRoot(root)
+	require.NoError(t, err)
+	require.Equal(t, ta.txh, txh)
+	require.Equal(t, ta.vub, vub)
+}
+
+func TestSetPrice(t *testing.T) {
+	ta := &testAct{}
+	nns := New(ta, util.Uint160{1, 2, 3})
+
+	priceList := []int64{100, 200}
+	_, err := smartcontract.NewParameterFromValue(priceList)
+	require.NoError(t, err)
```

Comment:
You'll receive error here because NewParameterFromValue doesn't handle  `[]int64`. And these two lines are not related to the test, remove them.